src/app/features/notification/notification.component.ts

```diff
@@ -0,0 +1,230 @@
+import { Component, OnInit } from "@angular/core";
+import { Subject, Subscription } from "rxjs";
+import { MatBadgeModule } from "@angular/material/badge";
+import { FontAwesomeModule } from "@fortawesome/angular-fontawesome";
+import { MatMenu, MatMenuModule, MatMenuTrigger } from "@angular/material/menu";
+import { MatButtonModule } from "@angular/material/button";
+import { FormsModule } from "@angular/forms";
+import { MatTooltipModule } from "@angular/material/tooltip";
+import { NotificationEvent } from "./model/notification-event";
+import { EntityMapperService } from "app/core/entity/entity-mapper/entity-mapper.service";
+import { MatTabsModule } from "@angular/material/tabs";
+import { NotificationItemComponent } from "./notification-item/notification-item.component";
+import { SessionSubject } from "app/core/session/auth/session-info";
+import { closeOnlySubmenu } from "./close-only-submenu";
+import { NotificationConfig } from "./model/notification-config";
+import { AlertService } from "app/core/alerts/alert.service";
+import { Router } from "@angular/router";
+import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
+import { applyUpdate } from "../../core/entity/model/entity-update";
+
+@UntilDestroy()
+@Component({
+  selector: "app-notification",
+  standalone: true,
+  imports: [
+    MatBadgeModule,
+    FontAwesomeModule,
+    MatMenu,
+    MatButtonModule,
+    MatMenuTrigger,
+    MatMenuModule,
+    FormsModule,
+    MatTooltipModule,
+    MatTabsModule,
+    NotificationItemComponent,
+  ],
+  templateUrl: "./notification.component.html",
+  styleUrl: "./notification.component.scss",
+})
+export class NotificationComponent implements OnInit {
+  public allNotifications: NotificationEvent[] = [];
+  public unreadNotifications: NotificationEvent[] = [];
+  private notificationsSubject = new Subject<NotificationEvent[]>();
+  public selectedTab = 0;
+  public hasNotificationEnabled = false;
+  protected readonly closeOnlySubmenu = closeOnlySubmenu;
+
+  constructor(
+    private entityMapper: EntityMapperService,
+    private sessionInfo: SessionSubject,
+    private alertService: AlertService,
+    private router: Router,
+  ) {}
+
+  ngOnInit(): void {
+    this.notificationsSubject.subscribe((notifications) => {
+      this.filterUserNotifications(notifications);
+    });
```

Comment:
why did you decide to work with a subject rather than a simple array?

Reply:
As per my understanding, I decided to use a `Subject` instead of a simple array because
1. The `Subject` enables pushing new notifications into the stream and ensures that all subscribers are updated instantly without additional logic to manually refresh the array.
2. Using a `Subject` helps centralize the flow of notifications. This ensures consistent behavior and minimizes the chances of missing updates, as all consumers rely on the same reactive data source.
3. `Subject` decouples the notification generation and consumption logic. For example, the component can subscribe to notifications and handle them as needed without directly modifying the underlying notification array.